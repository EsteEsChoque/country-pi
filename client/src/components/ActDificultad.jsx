import React from 'react';
const difficulties = [
    { value: "muyFacil", label: "Muy fácil" },
    { value: "facil", label: "Fácil" },
    { value: "normal", label: "Normal" },
    { value: "dificil", label: "Difícil" },
    { value: "experimentados", label: "Experimentados" },
    ];
// Componente de dificultad
const ActDificultad = ({ handleChange, formData }) => {
  return (
    <div className="dificultad">
      <label htmlFor="dificultad">Dificultad:</label><br/>
      {difficulties.map((difficulty) => (
          <div key={difficulty.value}>
            <input 
            type="radio" 
            id={`difficulty${difficulty.value}`} 
            name="difficulty" value={difficulty.value} 
            checked={formData.difficulty === difficulty.value} 
            onChange={handleChange}/>
            <label htmlFor={`difficulty${difficulty.value}`}>{difficulty.label}</label><br/>
        </div>
        ))}
    </div>
  );
};
export default ActDificultad;