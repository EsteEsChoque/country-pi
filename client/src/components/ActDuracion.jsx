import React from 'react';


const options = [];
for (let i = 0; i <= 6; i++) {
options.push(<option key={`${i}:00`} value={`${i}:00`}>{`${i}:00hs`}</option>);
options.push(<option key={`${i}:30`} value={`${i}:30`}>{`${i}:30hs`}</option>);
}
// Componente de dificultad
const ActDuracion = ({ handleChange, formData }) => {
  return (
    <div >
      <label htmlFor="duration">Duraci√≥n:</label><br/>
        <select 
        id="duration" 
        name="duration" 
        value={formData.duration} 
        onChange={handleChange}>
        {options}
        </select><br/>
    </div>
  )
};
export default ActDuracion;