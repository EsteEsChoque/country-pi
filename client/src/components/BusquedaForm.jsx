import React from "react";
import { useEffect, useState } from "react";
import { useDispatch, useSelector } from "react-redux";
import { paisFilter } from "../redux/actions";
import style from "./SegundoBuscador.module.css";
import { paisesSeleccionados1 } from "../redux/actions";

const BusquedaForm = ({ onCountryIdsChange,paisesSeleccionados,setPaisesSeleccionados }) => {
  const todopais = useSelector((state) => state);
  const dispatch = useDispatch();
  const [search, setSearch] = useState("");

  useEffect(() => {
    dispatch(paisFilter(search));
  }, [search]);
  const searcher = (e) => {
    if (e.target.value) {
      setSearch(e.target.value);
    }
  }

  

  const paisesMostrados = todopais.pais.slice(0, 7);

  const handleChange = (event) => {
    setSearch(event.target.value);
  };

  const handleAdd = ( pais) => {
    if (!paisesSeleccionados.includes(pais.id)) {
    setPaisesSeleccionados([...paisesSeleccionados, pais.id]);
    onCountryIdsChange([...paisesSeleccionados, pais.id]);
    } else {
    setPaisesSeleccionados([...paisesSeleccionados]);
    onCountryIdsChange([...paisesSeleccionados]);
    }
    };
    
    const handleRemove = (id) => {
    setPaisesSeleccionados(paisesSeleccionados.filter((p) => p !== id));
    onCountryIdsChange(paisesSeleccionados.filter((p) => p !== id));
    };
    
    return (
    <div>
    
    <div className={style.divForm}>
    <label htmlFor="searchActivi" className={style.label}>
    Buscar
    </label>
    <br />
    <input
           type="text"
           id="searchActivi"
           value={search}
           onChange={handleChange}
           placeholder="¿En qué paises?"
           className={style.input}
         />
             <div className={style.conteinerDiv}>
      <ul className={style.ul1}>
        {paisesMostrados.map((pais) => (
          <li key={pais.name} className={style.conteiner}>
            {pais.name}
            <img
              className={style.img}
              src={pais.img}
              alt={pais.name}
            />
            <button
              type="button"
              onClick={() => handleAdd(pais)}
              className={style.button}
            >
              ✓
            </button>
          </li>
        ))}
      </ul>
      <ul className={style.ul2}>
        {paisesSeleccionados.map((id) => {
        // Obtiene la información del país con el ID especificado
          const pais = todopais.paises.find((p) => p.id === id);
            return (
              <li key={pais.name} className={style.conteiner}>
              {pais.name}
                <img
                 className={style.img}
                 src={pais.img}
                 alt={pais.name}
               />
              <button
              type="button"
              onClick={() => handleRemove(id)}
              className={style.button}>X</button>
              </li>
            );
          })}
        </ul>
      </div>
    </div>
  </div>
);
};

export default BusquedaForm;

