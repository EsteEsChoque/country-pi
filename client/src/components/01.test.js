import React from 'react';
import ActDificultad from './ActDificultad';
import ActDuracion from './ActDuracion';

import BusquedaForm from './BusquedaForm';
import renderer from 'react-test-renderer';
import { screen, render,fireEvent } from '@testing-library/react';



test('ActDificultad se renderiza correctamente', () => {
  const handleChange = jest.fn();
  const formData = { difficulty: 'muyFacil' };
  const tree = renderer.create(<ActDificultad handleChange={handleChange} formData={formData} />).toJSON();
  expect(tree).toMatchSnapshot();
});

test('ActDuracion se renderiza correctamente', () => {
    const handleChange = jest.fn();
    const formData = { duration: '1:00' };
    render(<ActDuracion handleChange={handleChange} formData={formData} />);
  
    // Verifica que el menú desplegable se está renderizando correctamente
    const select = screen.getByRole('combobox');
    expect(select).toBeInTheDocument();
  
    // Verifica que el menú desplegable tiene las opciones correctas
    const options = screen.getAllByRole('option');
    expect(options).toHaveLength(14);
    expect(options[0]).toHaveTextContent('0:00hs');
    expect(options[1]).toHaveTextContent('0:30hs');
    // y así sucesivamente
  });
  
  
  
  