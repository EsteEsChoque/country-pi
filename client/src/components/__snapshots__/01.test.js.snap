// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`ActDificultad se renderiza correctamente 1`] = `
<div
  className="dificultad"
>
  <label
    htmlFor="dificultad"
  >
    Dificultad:
  </label>
  <br />
  <div>
    <input
      checked={true}
      id="difficultymuyFacil"
      name="difficulty"
      onChange={[MockFunction]}
      type="radio"
      value="muyFacil"
    />
    <label
      htmlFor="difficultymuyFacil"
    >
      Muy fácil
    </label>
    <br />
  </div>
  <div>
    <input
      checked={false}
      id="difficultyfacil"
      name="difficulty"
      onChange={[MockFunction]}
      type="radio"
      value="facil"
    />
    <label
      htmlFor="difficultyfacil"
    >
      Fácil
    </label>
    <br />
  </div>
  <div>
    <input
      checked={false}
      id="difficultynormal"
      name="difficulty"
      onChange={[MockFunction]}
      type="radio"
      value="normal"
    />
    <label
      htmlFor="difficultynormal"
    >
      Normal
    </label>
    <br />
  </div>
  <div>
    <input
      checked={false}
      id="difficultydificil"
      name="difficulty"
      onChange={[MockFunction]}
      type="radio"
      value="dificil"
    />
    <label
      htmlFor="difficultydificil"
    >
      Difícil
    </label>
    <br />
  </div>
  <div>
    <input
      checked={false}
      id="difficultyexperimentados"
      name="difficulty"
      onChange={[MockFunction]}
      type="radio"
      value="experimentados"
    />
    <label
      htmlFor="difficultyexperimentados"
    >
      Experimentados
    </label>
    <br />
  </div>
</div>
`;

exports[`Dificultad ActDificultad se renderiza correctamente 1`] = `
<div
  className="dificultad"
>
  <label
    htmlFor="dificultad"
  >
    Dificultad:
  </label>
  <br />
  <div>
    <input
      checked={true}
      id="difficultymuyFacil"
      name="difficulty"
      onChange={[MockFunction]}
      type="radio"
      value="muyFacil"
    />
    <label
      htmlFor="difficultymuyFacil"
    >
      Muy fácil
    </label>
    <br />
  </div>
  <div>
    <input
      checked={false}
      id="difficultyfacil"
      name="difficulty"
      onChange={[MockFunction]}
      type="radio"
      value="facil"
    />
    <label
      htmlFor="difficultyfacil"
    >
      Fácil
    </label>
    <br />
  </div>
  <div>
    <input
      checked={false}
      id="difficultynormal"
      name="difficulty"
      onChange={[MockFunction]}
      type="radio"
      value="normal"
    />
    <label
      htmlFor="difficultynormal"
    >
      Normal
    </label>
    <br />
  </div>
  <div>
    <input
      checked={false}
      id="difficultydificil"
      name="difficulty"
      onChange={[MockFunction]}
      type="radio"
      value="dificil"
    />
    <label
      htmlFor="difficultydificil"
    >
      Difícil
    </label>
    <br />
  </div>
  <div>
    <input
      checked={false}
      id="difficultyexperimentados"
      name="difficulty"
      onChange={[MockFunction]}
      type="radio"
      value="experimentados"
    />
    <label
      htmlFor="difficultyexperimentados"
    >
      Experimentados
    </label>
    <br />
  </div>
</div>
`;
